llvm/lib/Transforms/Instrumentation/InstrProfiling.cpp

```diff
@@ -1213,6 +1218,17 @@ Value *InstrLowerer::getBitmapAddress(InstrProfMCDCTVBitmapUpdate *I) {
 void InstrLowerer::lowerCover(InstrProfCoverInst *CoverInstruction) {
   auto *Addr = getCounterAddress(CoverInstruction);
   IRBuilder<> Builder(CoverInstruction);
+  if (ConditionalCounterUpdate) {
+    Instruction *SplitBefore = CoverInstruction->getNextNode();
+    auto &Ctx = CoverInstruction->getParent()->getContext();
+    auto *Int8Ty = llvm::Type::getInt8Ty(Ctx);
+    Value *Load = Builder.CreateLoad(Int8Ty, Addr, "pgocount");
+    Value *Cmp = Builder.CreateIsNotNull(Load, "pgocount.ifnonzero");
+    Instruction *ThenBranch =
+        SplitBlockAndInsertIfThen(Cmp, SplitBefore, false);
```

Comment:
I think this is ok now because of a recent change. See my comment https://github.com/llvm/llvm-project/pull/102542#discussion_r1712261570
But thanks for bringing this up!

Reply:
After reading my old comment again, I remembered that it's important that we compute the CFG hash before changing the CFG, otherwise IRPGO will not correctly attribute profile counts. Luckily we compute the CFG hash in the earlier pass `PGOInstrumentationGen`.

https://github.com/llvm/llvm-project/blob/c66dee4c6bd650ef20105532a311a95abb25ece5/llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp#L919-L923

My apologies, my original comment in https://github.com/llvm/llvm-project/pull/95585/files was wrong.

Reply:
IIUC, it is ok to add a block here, right? If that's the case, I'm going to resolve this conversation.

Reply:
I'll follow bitmap part in #110792.